smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/PaginatorGenerator.kt

```diff
@@ -122,22 +126,41 @@ class PaginatorGenerator : SwiftIntegration {
                 this.write(docBody)
             }
             writer.openBlock(
-                "public func \$LPaginated(input: \$N) -> \$N<\$N, \$N> {", "}",
+                "public func \$LPaginated(input: \$N) -> \$N<\$N, \$N> {",
+                "}",
                 operationShape.toLowerCamelCase(),
                 inputSymbol,
                 ClientRuntimeTypes.Core.PaginatorSequence,
                 inputSymbol,
-                outputSymbol
+                outputSymbol,
             ) {
-                writer.write(
-                    "return \$N<\$N, \$N>(input: input, inputKey: \\\$N.$markerLiteral, outputKey: \\\$N.$nextMarkerLiteral, paginationFunction: self.\$L(input:))",
-                    ClientRuntimeTypes.Core.PaginatorSequence,
-                    inputSymbol,
-                    outputSymbol,
-                    inputSymbol,
-                    outputSymbol,
-                    operationShape.toLowerCamelCase()
-                )
+                val isTruncatedFlag = outputShape
+                    .members()
+                    .firstOrNull { it.hasTrait(PaginationTruncationMember.ID) }
+                    ?.defaultName()
+
+                if (isTruncatedFlag != null) {
```

Comment:
Rather than an if-else for the entire return statement, can we construct a smaller string for the `isTruncatedKey:` param that is different whether or not the `isTruncated` flag is set?

This would save a lot of duplication

Reply:
oh yeah I tried that approach at first with StringBuilder and then reverted when it caused errors, Ill try again